### DISCUSSION TITLE
Is it possible to generate a where term that makes the where clause always return true, regardless of model?

### DISCUSSION THREAD

**Discussion by ropbastos** [prisma/prisma#18488] [Q&A] [answered]
My use case: I want to dynamically generate a term in where, that sometimes needs to "shut down" filtering - something like passing 

```
prisma.model.findMany({
    where: {
       1: 1
    }
}) 
```

Is something like that possible? 

Many thanks :)

**Comment by nurul3101** [comment, marked as answer]
Hello @ropbastos 👋 

If you want to "shut down" filtering, it would essentially mean that no filters should be applied. i.e. the query should return all records, right?

If that's the case, you can just remove the entire where clause to get all the records.

Something like this:

```typescript
prisma.model.findMany() 
```

Would this work for your use case?

**Reply by nurul3101** [reply]
@ropbastos Did you had a chance to try it out? Do let us know in case you are facing any issues.

**Reply by ropbastos** [reply]
Thanks for the response, I think that would certainly work if I change my code quite a bit, but let me try to explain why I was looking for a "always truthy" term:

My situation is that I have a function that dynamically generates where terms/clauses based on some inputs + user login data. 

It goes something like this:

```
prisma.model.findMany({
    where: {
       someFixedCondition: true,
       OR: [
           generateWhereCondition(user, someData),
           { someOtherColumn: false },
       ]  
    }
}) 
```

For some situations, it would be nice if "generateWhereCondition()" could return something that's always truthy, effectively ignoring the " { someOtherColumn: false }" check above, and making the OR evaluate to true regardless. I'm porting a codebase away from another ORM and the generateWhereCondition() is used in many places, and it currently takes advantage of the "1=1" trick to dynamically make some where terms always truthy, meaning, it ignores some checks.

**Reply by nurul3101** [reply]
Would this hacky workaround work for you?
```typescript
const results = await prisma.modelName.findMany({
  where: {
    id: {
      not: null
    }
  }
});
```
In this example, the where condition checks if the id field is not null. Since the id field is usually a primary key and cannot be null, this condition always returns true, and you'll get all the records from the table

**Reply by ropbastos** [reply]
Many thanks! I ended up using something similar. 

How do I mark this comment as an answer?

**Reply by nurul3101** [reply]
I am glad to hear that you were able to solve it 🙌
